libraries/heft-config-file/src/ConfigurationFileBase.ts

```diff
@@ -226,6 +319,14 @@ export interface IConfigurationFileOptionsBase<TConfigurationFile> {
    * configuration files.
    */
   propertyInheritanceDefaults?: IPropertyInheritanceDefaults;
+
+  /**
+   * Use this property if you need to validate the configuration file in ways beyond what JSON schema can handle.
+   * This function will be invoked after JSON schema validation.
+   *
+   * The function should throw if the configuration file is invalid.
```

Comment:
The documentation for `CustomValidationFunction` says it should return `false` if the configuration file is invalid.